crates/shared/src/price_estimation/competition.rs

```diff
@@ -450,7 +454,10 @@ mod tests {
         third
             .expect_estimate()
             .times(1)
-            .returning(move |_| async { Ok(estimate(3)) }.boxed());
+            .returning(move |_| async {
+                tokio::task::yield_now().await;
```

Comment:
This adjustment is a bit weird.
The new logic wraps `estimator.estimate()` in a future which results in this execution:
1. Start stage 2
2. build 2 futures like `async { (index, estimate().await) }`
3. start polling the futures
4. future of the `third` estimator gets polled to completion because it contains nothing actually `async`
5. racing estimator has enough `Ok` results to return early
6. test would fail because `forth.estimate()` was never called but 1 call was expected (remember that the call to `estimate()` is now wrapped in another future)

So to make the test pass we have to yield here.

Reply:
Again I'd put a short version of this as a comment in the code in case a future maintainer wonders. Alternatively we could first `.map(estimate)` and then enumerate and map the estimate result again (but that would change the production code to make the test work). Up to you.

Reply:
You could probably also assert hat the fourth is never called below instead of doing the extra yield.

Reply:
I ended up using `estimate().map()` so that you don't have to be aware of this detail when writing tests (also making the second comment unnecessary).
It makes the business logic very slightly less readable but in total I consider this the better option.